tests/devices/default_tensor/test_default_tensor.py

```diff
@@ -414,3 +417,41 @@ def circuit():
             return qml.expval(qml.Z(0))
 
         assert np.allclose(circuit(), 0.0)
+
+
+# At this stage, this test is especially relevant for the MPS method, but we test both methods for consistency.
+@pytest.mark.parametrize("num_orbitals", [2, 4])
+@pytest.mark.parametrize("method", ["mps", "tn"])
+def test_wire_order_dense_vector(method, num_orbitals):
+    """Test that the wire order is preserved if the initial state is created from a dense vector."""
+
+    dev = qml.device("default.tensor", wires=int(2 * num_orbitals + 1), method=method)
+    qubits = dev.wires.tolist()
+
+    wave_fun = np.random.random((2 ** (2 * num_orbitals))) + 1j * np.random.random(
+        (2 ** (2 * num_orbitals))
+    )
+    wave_fun = wave_fun / np.linalg.norm(wave_fun)
+
+    X0 = np.random.random((num_orbitals, num_orbitals))
+    U0 = expm((X0 + X0.T) / 2.0)
+
+    def basis_rotation_ops(unitary_matrix, wires):
+        _, givens_list = givens_decomposition(unitary_matrix)
+
+        for grot_mat, indices in givens_list:
+            theta = np.arccos(np.real(grot_mat[1, 1]))
+            qml.SingleExcitation(2 * theta, wires=[int(wires[indices[0]]), int(wires[indices[1]])])
```

Comment:
I'm guessing `indices` can be out of order here. Am I right?

Reply:
What do you mean by 'out of order'? Indices are used in line 444. For this test, we could probably exchange them and it would still be fine

Reply:
Isn't the goal of the PR is to handle things like `qml.StatePrep(state, wires=[2,3,1])`? So I'm wondering whether that is effectively covered by this test.

Reply:
If you run this test without the addition to the source code you get an error, so it seems to me that this is enough (at least at this stage). I wasn't even sure to implement a test in the first place since the source code addition is very small

Reply:
I see, all good then.